doc/nrf/migration/migration_guide_1.x_to_2.x.rst

```diff
@@ -0,0 +1,432 @@
+
+TF-M replaces SPM as the default secure firmware solution
+*********************************************************
+
+Trusted Firmware M (TF-M) replaces the Secure Partition Manager (SPM) for secure image firmware.
+TF-M is now enabled by default for most nRF9160 and nRF5340 applications and samples.
+
+
+*Required action*: To start using TF-M instead of SPM in your application, see the information in :ref:`ug_tfm_migrate`.
+
+Switch from TinyCBOR to zcbor
+*****************************
+
+With Zephyr v3.x.x, TinyCBOR becomes deprecated and replaced by zcbor.
+If you are using TinyCBOR to encode data into CBOR format in your application, follow this guide to migrate your code.
+
+.. note::
+   If you have extended MCUmgr with additional custom commands, then your code will not compile anymore as the commands will receive pointers to variables that no longer exist or that have changed their types during TinyCBOR to zcbor transition.
+
+.. _migration_tinycbor_to_zcbor_general:
+
+General migration
+=================
+
+The TinyCBOR to zcbor migration consists of two tasks:
+
+ 1. :ref:`migration_tinycbor_to_zcbor_general_init`
+ #. :ref:`migration_tinycbor_to_zcbor_general_api`
+
+The first task involves TinyCBOR decoding and encoding replacement, while the second task is function-to-function conversion, with a few exceptions in how parameters are passed and what return values are returned from API calls.
+
+.. _migration_tinycbor_to_zcbor_general_init:
+
+zcbor state initialization
+--------------------------
+
+You must first initialize the decoding and encoding state of zcbor.
+
+Decoding state initialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Decoding with TinyCBOR requires setting up a reader with a buffer by calling `cbor_buf_reader_init`, and attaching it to a parser state with `cbor_parser_init`.
+zcbor decoding is initialized with a call to `zcbor_new_decode_state` that directly ties the provided buffer to decoding state.
+For better understanding of how zcbor states work, refer to `zcbor documentation`_.
+The following is an example of simple zcbor state initialization for decoding:
+
+.. code-block:: c
+
+   #define N   20              /* Max expected encoded elements */
+   zcbor_state_t zs[2];
+   uint8_t buffer[SOME_SIZE];
+   /* Read some data to the buffer */
+   ...
+   zcbor_new_decode_state(zs, ARRAY_SIZE(zs), buffer, ARRAY_SIZE(buffer), N);
+   /* Decode data from the buffer with zs state */
+   ...
+
+In the above example, the zcbor state `zs` consists of two states, which is a minimal practical state count.
+These states will allow to work on flat CBOR encoding where no variables are embedded within lists or maps.
+Two states are used because one will be current state and the other can be used as a constant state where an encoding error is stored.
+In case you plan to have embedded lists or maps, the number of states for decoding must be increased by adding a number of levels of 2.
```

Comment:
```suggestion
In case you plan to have an embedded lists or maps, the number of states for decoding must be increased by adding a number of levels to the base 2; for example in case of three levels of map/list encoding the number of states should be set to 2 + 3.
```